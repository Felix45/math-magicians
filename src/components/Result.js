import React from 'react';
import PropTypes from 'prop-types';

class Result extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { total, next, operation } = this.props;
    const result = `${total || operation || next ? `${total || ''} ${operation || ''} ${next || ''}` : 0}`;
    return (
      <div className="bg-dark d-flex flex-end p-1">
        {result}
      </div>
    );
  }
}

Result.defaultProps = {
  total: 0,
  next: '',
  operation: '',
};

Result.propTypes = {
  total: PropTypes.number,
  next: PropTypes.string,
  operation: PropTypes.string,
};

export default Result;
