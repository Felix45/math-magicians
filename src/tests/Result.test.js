import renderer from 'react-test-renderer';
import Result from '../components/Result';

it('renders correctly', () => {
  const tree = renderer
    .create(<Result />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
