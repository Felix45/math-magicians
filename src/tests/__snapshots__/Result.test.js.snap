// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-dark d-flex flex-end p-1"
>
  0
</div>
`;
